packages/flutter/lib/src/material/tab_bar_theme.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'tabs.dart';
+
+/// Applies a theme to descendant [TabBar] widgets.
+///
+/// A tab bar theme describes the colors of the label and the size/shape of the
+/// [TabBar.indicator].
+///
+/// Descendant widgets obtain the current theme's [TabBarTheme] object using
+/// [ThemeData.tabBarTheme] on the [ThemeData] given by calling [Theme.of].
+/// [ThemeData.tabBarTheme] can be customized by copying it (using
+/// [TabBarTheme.copyWith]).
+///
+/// See also:
+///
+///  * [TabBar], Material Design widget that displays a horizontal row of tabs.
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class TabBarTheme extends Diagnosticable {
+  /// Creates a tab bar theme that overrides the defaults for the following
+  /// values for any [TabBar] in the context.
+  const TabBarTheme({
+    this.indicator,
+    this.indicatorSize,
+    this.labelColor,
+    this.unselectedLabelColor,
+  });
+
+  /// Default value for [TabBar.indicator] in this Theme.
+  final Decoration indicator;
+
+  /// Default value for [TabBar.indicatorSize] in this Theme.
+  final TabBarIndicatorSize indicatorSize;
+
+  /// Default value for [TabBar.labelColor] in this Theme.
+  final Color labelColor;
+
+  /// Default value for [TabBar.unselectedLabelColor] in this Theme.
+  final Color unselectedLabelColor;
+
+
+  /// Linearly interpolate between two tab bar themes.
+  ///
+  /// The arguments must not be null.
+  ///
+  /// The `t` argument represents position on the timeline, with 0.0 meaning
```

Comment:
It might be best if we used a @macro to define this monster.

Reply:
Done, although I'm not sure how to verify that I did it quite correctly.